### PR TITLE
Configmap directory

### LINKED ISSUE #405: ConfigMap can't handle directory
### Description
JKube can handle only file and not directory.
With kubernetes you can specify both file and directory with --from-file parameter. 

### Info

* Eclipse JKube version : 1.0.0
* Maven version (`mvn -v`) :  3.6.3
```

```
* Kubernetes / Red Hat OpenShift setup and version : 

* If it's a bug, how to reproduce :

* If it's a feature request, what is your use case :
When you have a directory with many files and you can't/won't specify all file names inside a custom yaml configuration
it would be better to have the possibility to specify only a directory inside yaml (with annotation) and having all regular files picked inside configmap with filename for key and content file for value

* Sample Reproducer Project : *[GitHub Clone URL]*

### PR DESCRIPTION
**Pull request by Danipiario**
Signed-off-by: Danipiario <daniele.pirola@icteam.it>

## Description
fix #405 

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [x] I have implemented unit tests to cover my changes
 - [x] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [x] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I tested my code in Kubernetes
 - [x] I tested my code in OpenShift

### CODE REVIEW

**Comment by rohanKanojia** [comment]
@Danipiario: Could you please add some documentation about this new feature?

**Comment by Danipiario** [comment]
I've add an example inside file _jkube_configmap_file.adoc, let me know if it's ok or if there is a better place for docs.
Sorry How can I squash previous commits?

**Comment by Danipiario** [comment]
I've squashed commits, at least I hope :-)

File: jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/ConfigMapEnricher.java [hunk 1]
```diff
@@ -124,12 +150,13 @@ private void addConfigMapFromXmlConfigurations(KubernetesListBuilder builder) {
...
                             name = Paths.get(file).getFileName().toString();
                         }
-                        addConfigMapEntryFromFile(configMapBuilder, name, file);
+                        addConfigMapEntryFromDirOrFile(configMapBuilder, name, file);
                     }
                 }
             }
 
-            if ((configMapBuilder.getData() != null && !configMapBuilder.getData().isEmpty()) || (configMapBuilder.getBinaryData() != null && !configMapBuilder.getBinaryData().isEmpty())) {
+            if (configMapBuilder.getData() != null && !configMapBuilder.getData().isEmpty()
```
Comment (manusa) [inline jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/ConfigMapEnricher.java:158]: Hi, thx for contributing and sorry for the belayed review, we were focused on 1.0.x release.

Why did we remove the parenthesis? Precedence is not that clear now.

Comment (Danipiario) [reply jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/ConfigMapEnricher.java:158]: Hi, it's a my IDE option that remove unecessary parenthesis. I can remove it if you want

Comment (manusa) [reply jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/ConfigMapEnricher.java:158]: Hi, yes please, leave it as it was. Intention looks clearer, I reviewed this fairly early in the morning and it took a while to figure out what was happening.

Comment (Danipiario) [reply jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/ConfigMapEnricher.java]: Parenthesis added

Comment (manusa) [reply jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/ConfigMapEnricher.java]: Thank you :pray:

**Review by manusa — APPROVED** [review]
Thank you very much for contributing, LGTM

**Review by rohanKanojia — APPROVED** [review]
Looks good overall, Thanks a lot!

Minor question, why is this file required?
`jkube-kit/enricher/generic/src/test/resources/test-dir/test-dir-nested/.gitkeep`

**Comment by Danipiario** [comment]
> Looks good overall, Thanks a lot!
> 
> Minor question, why is this file required?
> `jkube-kit/enricher/generic/src/test/resources/test-dir/test-dir-nested/.gitkeep`

It's for junit test. I test that the code handles only files and not dirs like kubernetes. So I create an empty dir but  with git at least one file is required inside the dir so the convention is to create a file .gitkeep

**Comment by rohanKanojia** [comment]
ohk, I see. Thanks a lot for your comment.

**Pull request merged**